app/controllers/support/providers/locations_check_controller.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Support
+  module Providers
+    class LocationsCheckController < SupportController
+      before_action :build_site_and_form
+
+      def show; end
+
+      def update
+        if @location_form.save!
+          if params.keys.include?('another')
+            redirect_to new_support_recruitment_cycle_provider_location_path
+          else
+            redirect_to support_recruitment_cycle_provider_locations_path
+          end
+          flash[:success] = t('support.providers.locations.added')
+        else
+          render template: 'support/locations/new'
+        end
+      end
+
+      private
+
+      def build_site_and_form
```

Comment:
is there a real need to build the site?

Reply:
I think so, we need to pass it to the form?

Reply:
```
@restores_from_store = location_form = LocationForm.new(provider)
@overwrite_from_store = LocationForm.new(provider, site)
```

Reply:
LocationForm.new(provider) raises an Argument error?

Reply:
Oh right, sorry was comparing it to `Publish::CourseFundingForm`